tests/integration/test_github.py

```diff
@@ -205,7 +205,7 @@ async def test_get_authenticated_user_no_refresh_token(self, codecov_vcr):
             "company": "@codecov ",
             "blog": "",
             "location": None,
-            "email": "thiago@codecov.io",
+            "email": None,
```

Comment:
Mmm is there a test where we actually get the email and verify it's public?

Reply:
I was also thinkin, how can we be certain we're still fetching the primary public email? 
Maybe we could check if the primary is public and return, otherwise find any public?

Reply:
> how can we be certain we're still fetching the primary public email

@ajay-sentry that's what we're doing! email in `/user` defaults to the fist public primary email (it's in the docs), if it doesn't exist we fetch the first public email in `/emails`. That's assuring us we're respecting the "visibility" of emails. (resources in the desc)

@adrian-codecov tbh i'm not sure, this fetches `https://github.com/login/oauth/authorize?response_type=code&scope=user%3Aemail%2Cread%3Aorg%2Crepo%3Astatus%2Cwrite%3Arepo_hook&client_id=999247146557c3ba045c` and tests it's values, i'm not sure how i can alter values in such case, any thoughts?

Reply:
Are you familiar w/ cassettes? 

The idea is that you make a real API call once, that creates a cassette, and then your test calls that cassette instead of calling the actual API. So you'd want to make a brand new test to an org you have a token of /access to. You'll need to use one the `*_handlers` , and replace the token with a real token/owner/repo combo for the test, and have that create a cassette. Does that make sense? Lmk and I can give you a hand otherwise

Reply:
https://sentry.slack.com/archives/C04M5GE2R5M/p1718885290037649

Reply:
It looks like we need to create a new github app to verify this